src/core-app-plane/bash-job-runner.ts

```diff
@@ -81,6 +81,12 @@ export interface BashJobRunnerProps {
    */
 
   readonly eventManager: IEventManager;
+
+  /**
+   * Any additional properties for the CodeBuild project beyond the default.
+   * These take precedence over defaults.
```

Comment:
I like the idea, but one concern I have is with regards to `props.scriptEnvironmentVariables`. With this new change, we'll have 2 parameters that can be used to set the env vars available in the codebuild project.

IMO, it would be better for us to reduce that to 1 by either:
- removing `props.scriptEnvironmentVariables`
- removing codeBuildProps and adding two new args: one for `buildImage` and one for `computeType`